source-stripe-native/source_stripe_native/api.py

```diff
@@ -0,0 +1,495 @@
+    fetch_incremental method. The only variation is that 
+    the resulting data serves has the search object for the next
+    stream (child stream). With that, a new iteration happens inside
+    this method exclusively for the child stream.
+    """
+
+    stop = True
+
+    url = f"{API}/events"
+    parameters = {"type": cls.TYPES, "limit": 100}
+    recent = []
+
+
+    _cls: Any = cls  # Silence mypy false-positive
+
+    while stop:
+        result = EventResult.model_validate_json(
+        await http.request(log, url, method="GET", params=parameters)
+    )
+        for results in result.data:
+            if _s_to_dt(results.created) > log_cursor:
+                recent.append(_s_to_dt(results.created))
+
+                parent_data = _cls.model_validate(results.data.object)
+                search_name = _cls.SEARCH_NAME
+                id = parent_data.id
+                child_data = _capture_substreams(
+                                    cls_child,
+                                    search_name,
+                                    id,
+                                    parent_data,
+                                    http,
+                                    log
+                                )
+
+                if child_data is None:
+                    pass # move to next customer
+                async for doc in child_data:
+                    doc.parent_id = parent_data.id
+                    doc.meta_ = cls_child.Meta(op="u")
+                    yield doc 
+        
+            elif _s_to_dt(results.created) < log_cursor:
+                stop = False
+                break
+        if result.has_more is True:
+            parameters["starting_after"] = result.data[-1].id
+        else:
+            break
+    if recent:
+        recent.sort()
+        yield recent[-1]
+
+async def fetch_backfill_substreams(
+    cls,
+    cls_child,
+    stop_date,
+    http: HTTPSession,
+    log: Logger,
+    page: str | None,
+    cutoff: datetime,
+)-> AsyncGenerator:
+    """Note: Stripe's data is always served in reverse-chronological order.
+    fetch_backfill_substreams works similar to fetch_backfill. The only variation is that 
+    the resulting data serves has the search object for the next
+    stream (child stream). With that, a new iteration happens inside
+    this method exclusively for the child stream.
+    """
+
+    _cls: Any = cls  # Silence mypy false-positive
+
+    url = f"{API}/{_cls.SEARCH_NAME}"
+    parameters = {"limit": 100}
+
+    search_name = _cls.SEARCH_NAME
+
+
+    if page:
+        parameters["starting_after"] = page
+    
+    if _cls.NAME == "Subscriptions" or cls.NAME == "SubscriptionItems":
+        parameters["status"] = "all"
+
+
+    
+    result = BackfillResult[_cls].model_validate_json(
+        await http.request(log, url, method="GET", params=parameters)
+    )
+
+    for doc in result.data:
+        log.debug(f"{_s_to_dt(doc.created)}")
```

Comment:
Does not seem like a useful log

Reply:
Yep, forgot to remove that one. Removed in 00467212d8135d9c433b5806262295c55c368f4c